app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Api::V1::BaseController < ActionController::Base
+include ActionController::HttpAuthentication::Basic::ControllerMethods
+include ActionController::HttpAuthentication::Token::ControllerMethods
+
+before_filter :authenticate_user_from_token, except: [:token]
```

Comment:
I think before_filter is deprecated. We might switch this to before_action sometime.